OpenTerm/Util/Cub.swift

```diff
@@ -0,0 +1,153 @@
+//
+//  Cub.swift
+//  OpenTerm
+//
+//  Created by Louis D'hauwe on 03/02/2018.
+//  Copyright © 2018 Silver Fox. All rights reserved.
+//
+
+import Foundation
+import Cub
+import ios_system
+
+public func cub(argc: Int32, argv: UnsafeMutablePointer<UnsafeMutablePointer<Int8>?>?) -> Int32 {
+
+    guard argc == 2 else {
+        fputs("Usage: cub script.cub\n", thread_stderr)
+        return 1
+    }
+    
+    guard let fileName = argv?[1] else {
+        fputs("Usage: cub script.cub\n", thread_stderr)
+        return 1
+    }
+    
+    let path = String(cString: fileName)
+
+    guard FileManager.default.fileExists(atPath: path) else {
+        fputs("Missing file \(path)\n", thread_stderr)
+        return 1
+    }
+    
+    let url = URL(fileURLWithPath: path)
+
+    guard let data = FileManager.default.contents(atPath: url.path) else {
+        fputs("Missing file \(path)\n", thread_stderr)
+        return 1
+    }
+
+    guard let source = String.init(data: data, encoding: .utf8) else {
+        fputs("Missing file \(path)\n", thread_stderr)
+        return 1
+    }
+    
+    let runner = Runner(logDebug: true, logTime: false)
+    
+    runner.registerExternalFunction(name: "print", argumentNames: ["input"], returns: false) { (arguments, callback) in
+        
+        for (name, arg) in arguments {
+            fputs("\(arg)\n", thread_stdout)
+        }
+        
+        callback(nil)
+        return
+    }
+
+    runner.registerExternalFunction(name: "exec", argumentNames: ["command"], returns: true) { (arguments, callback) in
+
+        var arguments = arguments
+
+        guard let command = arguments.removeValue(forKey: "command") else {
+            callback(.number(1))
+            return
+        }
+
+        guard case let .string(commandStr) = command else {
+            callback(.number(1))
+            return
+        }
+        
+        print("run command: \(commandStr)")
+
+        executor.dispatch(commandStr, callback: { (code) in
+
+            print("did run command: \(commandStr) -> \(code)")
+
+            DispatchQueue.main.async {
+                
+                callback(.number(Double(code)))
```

Comment:
Do we not care about the output of the command, just the return code?

Reply:
Maybe it would be better to create a class conforming to `CommandExecutorCommand`, then create your own `TerminalExecutor` and become its delegate, instead of using the visible view controller's executor.